### ISSUE TITLE
Problem with script in PHP CLI

### ISSUE THREAD

**Issue by mayaliny** [qtranslate/qtranslate-xt#766]
Hello, 
i'm writing a parser to be called from the command line, and i noticed that when qtranslate XT (ver. 3.7.1) is enabled the script simply exits (no errors shown or logged).

When the plugin is disabled the script is able to import data into WP correctly.

Are you doing something do disable the CLI?

FWIW I'm loading wordpress with this code at the very beginning of my script:

`// Let's load WordPress
   
    if( php_sapi_name() !== 'cli' ) {
        die("Meant to be run from command line");
    }

    function find_wordpress_base_path() {
        $dir = dirname(__FILE__);
        do {
            //it is possible to check for other files here
            if( file_exists($dir."/wp-config.php") ) {
                return $dir;
            }
        } while( $dir = realpath("$dir/..") );
        return null;
    }

    define( 'BASE_PATH', find_wordpress_base_path()."/" );
    define('WP_USE_THEMES', false);
    global $wp, $wp_query, $wp_the_query, $wp_rewrite, $wp_did_header;
    require(BASE_PATH . 'wp-load.php');
`

any clue?

**Comment by herrvigg** [comment]
I don't use this, but it is supposed to work. The only places where WP_CLI is handled is here to disabled the redirections:
https://github.com/qtranslate/qtranslate-xt/blob/1aeeee2bc95cea13e331560d050c3e073e1001c8/qtranslate_utils.php#L634

and here for the language detection:
https://github.com/qtranslate/qtranslate-xt/blob/1aeeee2bc95cea13e331560d050c3e073e1001c8/qtranslate_core.php#L148

Apart from this, nothing special. So if the scripts exits you should add more logging info to understand where it happens.

**Comment by hirasso** [comment]
Hi there! Today I also ran into a problem wit `WP_CLI` and qtranslate (I'm trying to implement a custom command). This is the error output I am getting:

```shell
Notice: Undefined index: wp-path in /.../content/plugins/qtranslate-xt/qtranslate_core.php on line 440
Warning: assert(): language needs to be shown in url - cancelled by can_redirect, url_info={
    "cookie_front_or_admin_found": false,
    "pagenow": "index.php",
    "REQUEST_METHOD": "GET",
    "scheme": "http",
    "host": "",
    "path": false,
    "path-base": "\/core",
    "doing_front_end": true,
    "lang_browser": null,
    "doredirect": "language needs to be shown in url - cancelled by can_redirect",
    "language": "de",
    "set_cookie": true
} failed in /.../plugins/qtranslate-xt/qtranslate_core.php on line 100
``` 
QTX_VERSION: 3.12.1
WP Version: 6.0.3

**Comment by herrvigg** [comment]
So WP_CLI is not set and the request looks really weird with `host="", path=false, path-base="\/core"`.
Can you explain what you are trying to do from the CLI?

**Comment by hirasso** [comment]
Maybe this issue is somehow related to my website using a custom folder structure? It looks like this:

```
.
├── content
├── core
├── index.php
└── wp-config.php
```
...where `/content` is my `wp-content` folder and `/core` contains all WP files.

By the way, I am also getting this error when trying to run any of the built-in WP CLI commands, like e.g. `wp core update`.

**Comment by herrvigg** [comment]
Related to #1326.

**Comment by herrvigg** [comment]
Fix released in 3.14.2.
